### PR TITLE
Add mimetype input

### PR DESCRIPTION
**Pull request by hrntsm**
## Motivation

Support for displaying 3d models in https://github.com/optuna/optuna-dashboard/pull/552
I am considering increasing the number of supported 3D file types.

Added an option to specify directly the mimetype of the 3d model file, since it is not always expected to be appropriate using the guess function.

For example, ".obj" file, which is one of the extensions of a 3D model, will be `application/x-tgif` in one environment, but `model/obj` in another environment.

This is because the function mimetypes.guess_type gets mimetype information from OS-dependent files. (I suspect this is due to the use of [knownfiles](https://github.com/python/cpython/blob/63a7f7765c6e9c1c9b93b7692e828ecf7bbd3bb9/Lib/mimetypes.py#L48) .)

## Description of the changes

The implementation was made the same as the [optuna dashboard implementation](https://github.com/optuna/optuna-dashboard/blob/941f926abdcff1fa7f9377cb8ce07937161f998f/optuna_dashboard/artifact/_backend.py#L142).

### CODE REVIEW

**Comment by c-bata** [comment]
@gen740 Could you review this PR?

File: optuna/artifacts/_upload.py
Comment (c-bata) [inline optuna/artifacts/_upload.py:38]: Could you update the `Args` section in the docstring?

**Review by c-bata — COMMENTED** [review]
@hrntsm Thank you for your pull request! Looks almost good to me. I left one comment though.

File: optuna/artifacts/_upload.py
Comment (c-bata) [inline optuna/artifacts/_upload.py:59]: A nit.

```suggestion
            An encoding of the artifact, which is suitable for use as a ``Content-Encoding``
            header (e.g. gzip). If not specified, the encoding is guessed from the file extension.
```

**Review by c-bata — APPROVED** [review]
LGTM with a nit! Thank you for your contribution.

File: optuna/artifacts/_upload.py
Comment (gen740) [inline optuna/artifacts/_upload.py:39]: Could you use `str | None` instead of `Optional`?

**Review by gen740 — APPROVED** [review]
Almost LGTM! I've left a minor comment. It seems something is wrong with the CI; could you please rerun it?

**Comment by c-bata** [comment]
Currently, integration tests seems to be broken since XGBoost 2.0.0 was released about 2 hours ago. I've just created a hotfix https://github.com/optuna/optuna/pull/4919, so please wait until #4919 will be merged :bow:

**Comment by c-bata** [comment]
@hrntsm Could you merge the main branch since #4921 was merged?

**Pull request merged**
